### PR TITLE
feat(mobile): multiselect for search & person page

### PR DESCRIPTION
**Pull request by fyfrey**
this PR enables multiselect on search result, person asset result and videos page.

It allows users to add assets from search to albums etc.

### CODE REVIEW

File: mobile/lib/modules/search/views/search_result_page.dart [hunk 1]
```diff
@@ -163,15 +154,15 @@ class SearchResultPage extends HookConsumerWidget {
...
-          );
-        }
+        return MultiselectGrid(
+          renderListProvider: searchRenderListProvider,
+          archiveEnabled: true,
+          deleteEnabled: true,
+          editEnabled: true,
+          favoriteEnabled: true,
+          stackEnabled: false,
+          onRefresh: refresh,
```
Comment (shenlong-tanwen) [inline mobile/lib/modules/search/views/search_result_page.dart:164]: This is great. Using the refresh in `MultiSelectGrid` to actually refresh the search

File: mobile/lib/modules/asset_viewer/providers/render_list.provider.dart [hunk 2]
```diff
@@ -16,6 +16,15 @@ final renderListProvider =
   );
 });
 
+final renderListProviderWithGrouping =
```
Comment (shenlong-tanwen) [inline mobile/lib/modules/asset_viewer/providers/render_list.provider.dart:19]: Nit. Maybe we can migrate the `renderListProviderWithGrouping` and the existing `renderListProvider` to the riverpod_generator and named argument for `GroupAssetsBy`? We can leave them as it is and can always migrate them later as well

**Review by shenlong-tanwen — APPROVED** [review]
LGTM

File: mobile/lib/modules/asset_viewer/providers/render_list.provider.dart (hunk 2 above)
Comment (fyfrey) [reply mobile/lib/modules/asset_viewer/providers/render_list.provider.dart:19]: let's migrate later :-)

**Pull request merged**